lib/common/output_text.c

```diff
@@ -328,3 +330,44 @@ pcmk__indented_printf(pcmk__output_t *out, const char *format, ...) {
     pcmk__indented_vprintf(out, format, ap);
     va_end(ap);
 }
+
+void
+pcmk__text_prompt(const char *prompt, bool echo, char **dest)
+{
+    int rc = 0;
+    struct termios settings;
+
+    if (!echo) {
+        rc = tcgetattr(0, &settings);
+        if (rc == 0) {
+            settings.c_lflag &= ~ECHO;
+            rc = tcsetattr(0, TCSANOW, &settings);
+        }
+    }
+
+    if (rc == 0) {
+        fprintf(stderr, "%s: ", prompt);
+
+        if (*dest != NULL) {
+            free(*dest);
```

Comment:
I'd also set *dest=NULL in case the scanf() fails, so it doesn't get double-freed